src/server/node_manager.c

```diff
@@ -7119,6 +7119,24 @@ set_nodes(void *pobj, int objtype, char *execvnod_in, char **execvnod_out, char
 				return (PBSE_UNKNODE);
 			}
 
+                        if ((pnode->nd_state & (INUSE_DOWN | INUSE_STALE)) && (svr_init == FALSE)) {
+
+				if(objtype == JOB_OBJECT) {
+                                   free(phowl);
+                                   free(execvncopy);
+                                   return (PBSE_BAD_NODE_STATE);
+				} 
+				else {
+					if(presv->ri_qs.ri_state == RESV_UNCONFIRMED) {
+						set_resv_retry(presv, time_now+10);
+						free(phowl);
+                	                	free(execvncopy);
+                        	      		return (PBSE_BAD_NODE_STATE);
+					}
+					else
+						resv_setResvState(presv, RESV_DEGRADED, RESV_DEGRADED);
```

Comment:
Speaking of reservations in the degraded state, look at the vnode_unavailable() function.  That is the function that usually marks reservations as degraded.  It sets the retry time.  I think just setting the state will mark the reservation in state degraded, but the scheduler will never reconfirm it.

Reply:
For Standing reservation, if state is set to degraded it will be reconfirmed by scheduler by setting retry time.

Reply:
Yes, that is my point.  You are not setting the retry time.  You are just setting the state.